### ISSUE TITLE
Decide on a license for the repo

### ISSUE THREAD

**Issue by mjskay** [transparentstats/guidelines#15]
Some obvious options:

MIT
Apache
GPL
CC-BY

See e.g. [here](https://choosealicense.com/)

I lean towards MIT or Apache as they are permissive, so if people want to use code examples in their own analyses and put their code under a different license they can, but they still have to acknowledge where it came from. To me that precludes using GPL. CC-BY seems like another good choice, although they seem to [recommend against using it for code](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software), so that leads me back to Apache or MIT due to our use of code in exemplars.

The main difference between Apache and MIT seems to be that Apache also has a clause granting patent rights from contributors to users (again, see [here](https://choosealicense.com/)). I'm not really sure that makes a big difference for us.

Once we decide, I will add a LICENSE file to the repo.

@chatchavan @steveharoz @dragice @shionguha @patrickgage Thoughts?

**Comment by shionguha** [comment]
I'll vote for the MIT license. I base my decision on [https://choosealicense.com/](url) and then by further reading [https://choosealicense.com/licenses/mit/](url)

**Comment by mjskay** [comment]
Followup question: The licenses typically include a bit that lists who the copyright belongs to, like this:

```
Copyright (c) [year] [fullname]
```

Do we want to keep track of all contributor names there? That might get painful (though maybe with github its actually not hard to track...). I've seen some projects that list core people and then say "and contributors", which is another option.

**Comment by chatchavan** [comment]
Can we use both? CC-BY for documentation and MIT for code. If somebody else use the guideline text (e.g., in a textbook), I'd like them to at least give attribution to the repository. Just so that readers (of the textbook) has a pointer to the latest version of the guideline.

**Comment by mjskay** [comment]
Oh that's a good point. In that case do we want CC-BY or CC-BY-SA for the text? I am now thinking it would be good to look at some other online textbooks, like those on bookdown.org, to see how they do this.

**Comment by steveharoz** [comment]
I vote CC-BY. I'm generally not a fan of SA (share alike), as I prefer not to limit how others use their adaptations.

**Comment by mjskay** [comment]
Now I am hesitating to go straight CC-BY, since that means someone could basically take the content and print a book if they wanted to. For what it is worth, all of the books on the front page of bookdown.org that I can determine the license of are one of the more restrictive variants of CC-BY, like CC-BY-NC-SA (attribution + non-commercial + sharealike) or CC-BY-NC-ND (ND = no derivatives).

Steve, I understand your point with respect to code examples, which is why I think a permissive license just for code blocks might be a good idea. However, I'm trying to find an example of someone having done this. It actually strikes me as problematic that there are a bunch of programming books and documentation out there with licenses like CC-BY-SA (or even especially -ND) since I would think that implied the code examples could only be used in sharealike compatible software (basically, GPL'd software) or, in the case of no derivatives, could not be used at all. Maybe I'm misunderstanding some subtle point of copyright law that makes this all okay and nothing to worry about; I have to guess that is it because I find it hard to believe some of the people writing these books have not thought about this.

I might ask some of the people at Michigan who have been working on some online textbooks if they have insight into this.

**Comment by dragice** [comment]
I'm all for a CC license for both code and documents. Concerning the distinction between CC-BY and CC-BY-SA, this page seems to clarify:

https://wiki.creativecommons.org/wiki/ShareAlike_interpretation

> **The ShareAlike condition only applies when a work is publicly shared.** You are not obligated to share things you make from SA works--you may create remixes and adaptations that you do not publish. If you are using ShareAlike materials privately and not sharing them with others, you do not have to comply with the license conditions. For example, if you translate a ShareAlike work for internal use within your office, you do not have to license your translation under an SA or compatible license unless you plan to share it with others.

**Comment by chatchavan** [comment]
Just wrote an email asking if there's any people at my university that I can consult.

Below is the email that includes the summary of our requirements (in case you need to brief people whom you'll consult).


> Dear colleagues,
> 
> Do you know if UZH has a specialist in copyright practice that I can consult? 
> 
> Do any of you have experience about open licenses such as CC-BY-SA?
> 
> 
> **Context:** My colleagues in HCI community and I are drafting a guideline for practicing transparent statistics. This guideline aims to improve the quality of statistical information presented in scientific articles. It targets authors (explaining desirable practices + providing example code) and reviewers (explaining rules of thumb in evaluating the quality of statistical report). We plan to release several versions as the knowledge about statistical practices grows from HCI, InfoVis and statistics community.
> 
> We wish to make the guideline as permissive as possible for widespread use. E.g., we won't mind if somebody copy and paste the code to use in their data analysis and publish the code in the supplement material of their article. We also don't mind if the guideline and the code will be used in lecture slides, blogs, etc. We wish that if the guideline was used in educational materials, such materials should point back to the online repository of the guideline so that readers are aware that an updated version may exist (and they should use the updated version).
> 
> We want to prevent somebody may just print out the whole content and publish it without acknowledging us.
> 
> We are considering CC-BY or CC-BY-SA. Our discussion thread about license type is here.
> 
> Cheers,
> Chat

**Comment by chatchavan** [comment]
I scheduled to discuss with a copyright/license specialist at my university on Monday 13:30, Zurich time. If there's any other questions or info that I should know, please add it to this issue before then.

**Comment by chatchavan** [comment]
Here's the result from the discussion about license:

* If we plan to publish the content, e.g., as a book chapter, we should check with the publisher before deciding on the license type. 
   * Many traditional publishers won't publish the content that is already under open licenses. 
   * Other open publishers will publish only if the original content is under the non-commercialized license.
* CC licenses doesn't cover software (including source code). Thus, we have to specify two separate licenses: one for the FAQ and another for the exemplar.
   * This can easily be done in a README or Table of Content file. Just add a statement clarifying both licenses.
* **License for FAQ**
   * CC-BY-SA is usually useful if we plan to incorporate materials from the derivatives back into our work. Otherwise, it's usually more restrictive for others to use our work.
   * CC-BY would be the best type for disseminating widely.
* **License for code**: MIT license is the best type for disseminating widely with least restriction.

**Comment by mjskay** [comment]
Thanks Chat, this is helpful.

> Other open publishers will publish only if the original content is under the non-commercialized license.

This may explain why people on bookdown.org seem to be using NC variants of CC-BY.

More broadly, I'm still somewhat confused, and I think part of the problem is I don't have a clear sense of what we want to support and what we don't. We might be putting the cart (license) before the horse (our needs). So I suggest we go at this the other way around: enumerate a concrete set of activities we want to support and don't want to support. Then perhaps we can take those scenarios to an expert (like Chat's) and say: "give us the right license for this!"


As a starting point, I think we should support:

1. People taking exemplar templates (text and code; i.e. Rmd files) and dropping their data into them and then releasing their data with the modified Rmd files. When doing this they should give us attribution. [Insert something here about licensing: do we want to force them to also open up their analysis under a similar license?]

2. Us taking the content and publishing it as a book.


Not support:

1. Someone else taking the content and publishing it as a book.


Questionable:

1. Relicensing in the future? (i.e. do we need some kind of contributor agreement that assigns copyright to us if we want to change terms in the future?)

**Comment by steveharoz** [comment]
If people use an exemplar template and fill in their own data, would CC-BY-SA prevent them from publishing in a non-open-access journal/conference?

**Comment by chatchavan** [comment]
@steveharoz This depends on the policy of the journal/conference.

**Comment by mjskay** [comment]
Ah, good point. I suppose if they had a policy on the copyright for supplemental material published with the paper. I am assuming CC-BY-SA would not require them to license the whole paper under a similar license, only the supplement if they wanted to publish the Rmd... Maybe that is optimistic though. Journals are often dicks.

Anyway I see the point --- the larger goal is better analyses, and hindering that with stricter licenses forcing open data is probably the wrong approach. So we have:

Support:

1. People taking exemplar templates (text and code; i.e. Rmd files) and dropping their data into them and then releasing their data with the modified Rmd files. When doing this they should give us attribution. ~~[Insert something here about licensing: do we want to force them to also open up their analysis under a similar license?]~~

2. People being able to incorporate analyses into papers, even if submitted to a non-open-access journal / a journal that wants to take the copyright.

3. Us taking the content and publishing it as a book.

Not support:

1. Someone else taking the content and publishing it as a book. [May have to compromise on this to support previous goals]

Questionable:

2. Relicensing in the future? (i.e. do we need some kind of contributor agreement that assigns copyright to us if we want to change terms in the future?)

**Comment by chatchavan** [comment]
Results from a consultation with the copyright specialist (numbered according to Matt's comment above):


We wish to support:
1. People taking exemplar templates (text and code; i.e. Rmd files) and dropping their data into them and then releasing their data with the modified Rmd files. When doing this they should give us attribution. 

> This is ensured with any CC license that allows modifications (such as CC-BY), and any open source license (such as MIT).

2. People being able to incorporate analyses into papers, even if submitted to a non-open-access journal / a journal that wants to take the copyright. 

> This depends on the journal. Generally, this should be possible with any CC license that allows modifications (such as CC-BY)

3. Us taking the content and publishing it as a book. 

> This depends on the publisher. You will have to make sure that he allows publishing of the content also under CC-BY in parallel.

We wish to *not* support:

4. Someone else taking the content and publishing it as a book. 

> If you release it under a CC license that allows modifications, you can not really prevent this. 


I haven't got the answer for point 5. (re-licensing)


----

Bottom line: I'd interpret this as CC-BY for text and MIT for code

**Comment by mjskay** [comment]
Okay, sounds good to me. 

It occurs to me that the text and code are integrated in some places (like inline code in suggested paper snippets). Does that complicate things? Should we say that the whole thing is under CC-BY, and that code snippets are additionally licensed under MIT?

**Comment by chatchavan** [comment]
Yes. Technically, CC-BY doesn't cover the code anyway. :)

Cheers,
Chat

**Comment by mjskay** [comment]
#35 closes this.
